src/main/scala/chiseltest/internal/ThreadedBackend.scala

```diff
@@ -188,18 +199,11 @@ class ThreadedBackend(annotations: AnnotationSeq) extends BackendInterface with
           (s"$actual (${bigintToHex(actual)})", s"$value (${bigintToHex(value)})")
       }
       val trace = new Throwable
-      val expectStackDepth = trace.getStackTrace.indexWhere(ste =>
-        ste.getClassName == "chiseltest.package$testableData" && ste.getMethodName == "expect"
-      ) + 1
-      require(
-        expectStackDepth != 0,
-        s"Failed to find expect in stack trace:\r\n${trace.getStackTrace.mkString("\r\n")}"
+      val fullStackTraces = (trace.getStackTrace ++ getParentTraceElements).filterNot(ste =>
+        ste.getClassName.startsWith("chiseltest.internal.")
```

Comment:
This significantly changes the behavior - this is filtering out, instead of filtering all before or after. The criteria is also looser. Both are subtle, but makes the code more brittle.

Yes, I know that under the old one, changes elsewhere means this also needs to be changed too. But we have tests to catch that, and I'd prefer code errs on the stricter and safer side.